elastic/kibana
# [Spaces] Add solution view select in UI

In this PR I've added a new "Navigation" section to the Create/Edit space UI. This section will allow to set the solution view ("search", "observability", "security" or "classic") for a specific space.

The initial value for the `solution` on the space is `undefined` which renders the "classic" navigation. This allows us to distinguis between spaces where a user has specifically asked for the "classic" navigation and spaces where no users action has been taken (in that case we could invite the user to set the solution view).

**Important:** Although this PR is targetting `main`, the new UI is behind a feature flag and will be enabled for `8.16`. This allows us to start testing in QA env.

## How to test

* Add the following in `kibana.dev.yml`

```yml
xpack.cloud.id: "ftr_fake_cloud_id:aGVsbG8uY29tOjQ0MyRFUzEyM2FiYyRrYm4xMjNhYmM="
xpack.cloud.base_url: "https://cloud.elastic.co/"
xpack.cloud.deployment_url: "/deployments/deploymentId"
xpack.cloud_integrations.experiments.enabled: true
xpack.cloud_integrations.experiments.flag_overrides:
  "solutionNavEnabled": true
```

* Go to management / spaces and create a new space
* You should see the new "Navigation" section
* Turn off the flag in the `yml` settings, refresh the browser and check that the section is not visible anymore

## Screenshots

<img width="1257" alt="Screenshot 2024-06-18 at 11 20 27" src="https://github.com/elastic/kibana/assets/2854616/1cc7eeb3-6f75-4561-bc6d-9c746ff7a0c8">
<img width="1271" alt="Screenshot 2024-06-18 at 11 20 37" src="https://github.com/elastic/kibana/assets/2854616/76961292-09a9-4346-b683-ebd81cd0d2d1">
<img width="1282" alt="Screenshot 2024-06-18 at 11 20 57" src="https://github.com/elastic/kibana/assets/2854616/78b61454-7d5a-4f95-af57-e7e76722d2b9">
